src/Neo/Cryptography/Crypto.cs

```diff
@@ -50,22 +51,30 @@ public static byte[] Hash256(ReadOnlySpan<byte> message)
         }
 
         /// <summary>
-        /// Signs the specified message using the ECDSA algorithm.
+        /// Signs the specified message using the ECDSA algorithm and specified hash algorithm.
         /// </summary>
         /// <param name="message">The message to be signed.</param>
         /// <param name="priKey">The private key to be used.</param>
         /// <param name="ecCurve">The <see cref="ECC.ECCurve"/> curve of the signature, default is <see cref="ECC.ECCurve.Secp256r1"/>.</param>
+        /// <param name="hasher">The hash algorithm to hash the message, default is SHA256.</param>
         /// <returns>The ECDSA signature for the specified message.</returns>
-        public static byte[] Sign(byte[] message, byte[] priKey, ECC.ECCurve ecCurve = null)
+        public static byte[] Sign(byte[] message, byte[] priKey, ECC.ECCurve ecCurve = null, Hasher hasher = Hasher.SHA256)
```

Comment:
I think the default will not be good.
But let other comment as well.

Personally, I prefer to pass the hash in every call.

Reply:
Got your idea, I have no special preferences, let's wait for other opinions.

Reply:
This just follows current default, I'd expect it to be used in the vast majority of cases, so default might be handy. But not critical to me either.

Reply:
> I think the default will not be good. But let other comment as well.
> 
> Personally, I prefer to pass the hash in every call.

Making this default ensures that we dont need to update other code or other repo that is not related to this pr.

Reply:
> we dont need to update other code or other repo

Good point, thus, let's keep the default value.

Reply:
I still think this should be marked as `[Obsolete]` and creating of new method. We dont want to break wallets, dapps, etc.

Reply:
Ref. https://github.com/neo-project/neo/pull/3209#discussion_r1590718313. BTW, you may also check that no changes are required from the user-side by looking at https://github.com/neo-project/neo-devpack-dotnet/pull/1035 as an example.